File: 1528-shuffle-string/1528-shuffle-string.java
class Solution {
    public String restoreString(String s, int[] indices) {
        char[] arr = new char[s.length()];

        for(int i = 0; i < s.length(); i++)
            arr[indices[i]] = s.charAt(i);
        String result = new String(arr);
        return result;
    }
}

File: 1528-shuffle-string/1528-shuffle-string.c
char* my_strcpy(char *dest, char *src)
{
    int i = 0;
    
    while(src[i])
    {
        dest[i] = src[i];
        i++;
    }
    dest[i] = '\0';
    return (dest);
}

char * restoreString(char * s, int* indices, int indicesSize){
    int j = 0;
    char tmp;
    char k[128];
    my_strcpy(k,s);
    
    while (j < indicesSize)
    {
        tmp = k[j];
        k[j] = s[indices[j]];
        s[indices[j]] = tmp;
        j++;
    }
    return s;
}